controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -429,10 +440,9 @@ func validateCoreDNSImageTag(fromTag, toTag string) error {
 	if err != nil {
 		return errors.Wrapf(err, "failed to parse CoreDNS target version %q", toTag)
 	}
-	// make sure that the version we're upgrading to is greater than the current one,
-	// or if they're the same version, the raw tags should be different (e.g. allow from `v1.17.4-somevendor.0` to `v1.17.4-somevendor.1`).
-	if x := from.Compare(to); x > 0 || (x == 0 && fromTag == toTag) {
-		return fmt.Errorf("toVersion %q must be greater than fromVersion %q", toTag, fromTag)
+	// make sure that the version we're upgrading to is greater or equal to the current one,
```

Comment:
Why are we changing this? Won't this lead to issues?

Reply:
See https://github.com/kubernetes-sigs/cluster-api/pull/6917#discussion_r922006323 ,

TLDR: if we don't allow keeping the same tag it would not result in a rollout / changing the image repository for coredns.